<div style="
    width: 80%;
    margin: 50px auto;
    padding: 45px 30px;
    border: 1px solid #555;
    border-radius: 6px;
    text-align: center;
    background: linear-gradient(to bottom, #ffffff, #f5f5f5);
    box-shadow: 0 4px 12px rgba(0,0,0,0.12);
">

<div style="
    font-size: 14px;
    letter-spacing: 3px;
    margin-bottom: 18px;
    color: #666;
">
DATA ANALYSING PROJECT
</div>

<h1 style="
    margin: 0;
    font-size: 30px;
    line-height: 1.4;
    font-weight: 700;
    color: #222;
">
Impact of Artificial Intelligence<br>
on Students
</h1>

<div style="
    width: 55px;
    height: 2px;
    background: #555;
    margin: 25px auto 0;
">
</div>

</div>

<center>
    <img src="ai_student.jpeg" width="700">
</center>

<div style="
    background-color: #f8f9fa;
    border-left: 5px solid #333;
    border-radius: 10px;
    padding: 20px 25px;
    margin: 25px 0;
    box-shadow: 0 2px 8px rgba(0,0,0,0.08);
">
       <h2 style="
        text-align: center;
        font-size: 24px;
        margin-bottom: 15px;
        color: #222;
    ">
        Introduction
    </h2>
        <p style="
        font-size: 16px;
        line-height: 1.8;
        text-align: justify;
        color: #444;
    ">
        Artificial Intelligence (AI) has become an important part of
        modern education. Students use AI tools for learning, research,
        assignments, problem-solving, and improving their academic skills.
        This project focuses on understanding the impact of AI on students
        and how the use of AI influences their learning experience and
        academic performance. The dataset contains information related to
        students and their use of AI, which can be explored to identify
        useful patterns and insights.
    </p>

</div>

<div style="
    background-color: #f8f9fa;
    border-left: 5px solid #333;
    border-radius: 10px;
    padding: 20px 25px;
    margin: 25px 0;
    box-shadow: 0 2px 8px rgba(0,0,0,0.08);
">
       <h2 style="
        text-align: center;
        font-size: 24px;
        margin-bottom: 15px;
        color: #222;
    ">
Abstract
</h2>

<p style="
    text-align: justify;
    font-size: 16px;
    color: #333;
">
Artificial Intelligence (AI) has become an important part of students’
academic life. Generative AI tools are increasingly used for learning,
assignments, research, writing and problem-solving. This project focuses on
understanding students’ AI usage and exploring its relationship with their
study habits, academic performance and overall academic experience.
</p>

<p style="
    text-align: justify;
    font-size: 16px;
    color: #333;
">
The dataset contains 50,000 student records covering AI usage, weekly
Generative AI hours, primary use cases, prompting skills, AI dependency,
traditional study hours, pre- and post-semester GPA, examination anxiety,
skill retention and burnout risk. The main objective is to identify useful
patterns in AI usage and examine its association with different aspects of
students’ academic life.
</p>

<p style="
    text-align: justify;
    font-size: 16px;
    color: #333;
">
The project uses Python and Pandas for data inspection, cleaning and basic
data analysis. Missing values, duplicates, data types and consistency are
checked before analysing the data using basic Pandas operations. The findings
help describe the observed patterns between AI usage, academic performance,
study habits and student well-being.
</p>

</div>

<div style="
    background-color: #f8f9fa;
    border-left: 5px solid #333;
    border-radius: 10px;
    padding: 20px 25px;
    margin: 25px 0;
    box-shadow: 0 2px 8px rgba(0,0,0,0.08);
">
       <h2 style="
        text-align: center;
        font-size: 24px;
        margin-bottom: 15px;
        color: #222;
    ">
About The DataSet
</h2>

<p style="
    text-align: justify;
    font-size: 16px;
    color: #333;
">
The <b>Impact of AI on Students</b> dataset contains information about
students and their use of Generative Artificial Intelligence in academic
activities. It provides data on AI usage, study habits, academic performance,
AI dependency and other factors related to students’ academic experience.
</p>

<p style="
    text-align: justify;
    font-size: 16px;
    color: #333;
">
The dataset contains <b>50,000 student records and 16 columns</b>. It includes
information such as weekly Generative AI usage, primary use cases, prompting
skills, AI tool diversity, traditional study hours, pre- and post-semester
GPA, examination anxiety, skill retention and burnout risk.
</p>

<p style="
    text-align: justify;
    font-size: 16px;
    color: #333;
">
This dataset is used in this project to perform <b>data cleaning and basic
data analysis</b>. The analysis aims to explore patterns in AI usage and its
association with students’ study habits, academic performance and other
academic-related factors.
</p>

</div>

<div style="
    width: 92%;
    margin: 35px auto;
    padding: 30px;
    border: 1px solid #555;
    border-radius: 8px;
    background-color: #ffffff !important;
    color: #000000 !important;
">

<h2 style="
    text-align: center;
    margin-top: 0;
    margin-bottom: 25px;
    font-size: 25px;
    letter-spacing: 1px;
    color: #000000 !important;
">
Column Description
</h2>

<table style="
    width: 100%;
    border-collapse: collapse;
    font-size: 14px;
    background-color: #ffffff !important;
    color: #000000 !important;
">

<tr style="
    background-color: #ffffff !important;
    color: #000000 !important;
">
    <th style="border: 1px solid #999; padding: 12px; background-color: #ffffff !important; color: #000000 !important;">Column Name</th>
    <th style="border: 1px solid #999; padding: 12px; background-color: #ffffff !important; color: #000000 !important;">Raw Type</th>
    <th style="border: 1px solid #999; padding: 12px; background-color: #ffffff !important; color: #000000 !important;">Target Type</th>
    <th style="border: 1px solid #999; padding: 12px; background-color: #ffffff !important; color: #000000 !important;">Description</th>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Student_ID</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">int64</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Integer</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Unique identification number assigned to each student.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Major_Category</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Major or field of study of the student.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Year_of_Study</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Current year of study of the student.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Pre_Semester_GPA</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">float64</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Numeric</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">GPA recorded before the semester.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Weekly_GenAI_Hours</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">float64</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Numeric</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Number of hours spent using Generative AI per week.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Primary_Use_Case</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Main purpose for which the student uses AI tools.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Prompt_Engineering_Skill</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Level of skill in creating effective prompts for AI tools.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Tool_Diversity</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">int64</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Numeric</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Number of different AI tools used by the student.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Paid_Subscription</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">bool</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Boolean</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Indicates whether the student has a paid AI subscription.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Traditional_Study_Hours</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">float64</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Numeric</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Number of hours spent on traditional study activities.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Perceived_AI_Dependency</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">int64</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Numeric</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Student's perceived level of dependency on AI tools.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Institutional_Policy</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">AI usage policy followed by the student's institution.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Anxiety_Level_During_Exams</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">int64</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Numeric</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Level of anxiety experienced by the student during examinations.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Post_Semester_GPA</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">float64</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Numeric</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">GPA recorded after the semester.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Skill_Retention_Score</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">float64</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Numeric</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Score representing the student's retention of learned skills.</td>
</tr>

<tr>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Burnout_Risk_Level</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">object</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Categorical</td>
<td style="border:1px solid #bbb;padding:10px;background-color:#ffffff !important;color:#000000 !important;">Level of risk of experiencing academic burnout.</td>
</tr>

</table>
</div>

### IMPORT PANDAS & LOAD THE DATASET

In [2]:
import pandas as pd
import warnings
warnings.filterwarnings("ignore")

In [3]:
df=pd.read_csv("ai_impact_students.csv")
df.head()

,Student_ID,Major_Category,Year_of_Study,Pre_Semester_GPA,Weekly_GenAI_Hours,Primary_Use_Case,Prompt_Engineering_Skill,Tool_Diversity,Paid_Subscription,Traditional_Study_Hours,Perceived_AI_Dependency,Institutional_Policy,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score,Burnout_Risk_Level
0,100001,Humanities,Senior,2.418,23.31,Copywriting/Drafting,Beginner,1,True,8.13,5,Allowed_With_Citation,6,2.393,86.44,High
1,100002,Medical,Junior,3.821,1.12,Ideation,Advanced,5,False,16.65,3,Allowed_With_Citation,9,3.696,69.39,Low
2,100003,Business,Freshman,3.398,21.26,Summarizing_Reading,Beginner,2,False,10.35,5,Strict_Ban,9,3.499,73.93,Medium
3,100004,Business,Senior,3.789,1.82,Copywriting/Drafting,Intermediate,4,False,15.23,2,Allowed_With_Citation,2,4.000,63.58,Medium
4,100005,STEM,Sophomore,3.635,9.29,Debugging/Troubleshooting,Advanced,4,False,12.55,4,Allowed_With_Citation,4,3.798,100.00,Medium


### DATASET OVERVIEW

In [4]:
df.shape

(50000, 16)

In [5]:
df.columns

Index(['Student_ID', 'Major_Category', 'Year_of_Study', 'Pre_Semester_GPA',
       'Weekly_GenAI_Hours', 'Primary_Use_Case', 'Prompt_Engineering_Skill',
       'Tool_Diversity', 'Paid_Subscription', 'Traditional_Study_Hours',
       'Perceived_AI_Dependency', 'Institutional_Policy',
       'Anxiety_Level_During_Exams', 'Post_Semester_GPA',
       'Skill_Retention_Score', 'Burnout_Risk_Level'],
      dtype='object')

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 16 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Student_ID                  50000 non-null  int64  
 1   Major_Category              50000 non-null  object 
 2   Year_of_Study               50000 non-null  object 
 3   Pre_Semester_GPA            50000 non-null  float64
 4   Weekly_GenAI_Hours          50000 non-null  float64
 5   Primary_Use_Case            50000 non-null  object 
 6   Prompt_Engineering_Skill    50000 non-null  object 
 7   Tool_Diversity              50000 non-null  int64  
 8   Paid_Subscription           50000 non-null  bool   
 9   Traditional_Study_Hours     50000 non-null  float64
 10  Perceived_AI_Dependency     50000 non-null  int64  
 11  Institutional_Policy        50000 non-null  object 
 12  Anxiety_Level_During_Exams  50000 non-null  int64  
 13  Post_Semester_GPA           500

In [7]:
df.describe()

,Student_ID,Pre_Semester_GPA,Weekly_GenAI_Hours,Tool_Diversity,Traditional_Study_Hours,Perceived_AI_Dependency,Anxiety_Level_During_Exams,Post_Semester_GPA,Skill_Retention_Score
count,50000.000000,50000.000000,50000.000000,50000.00000,50000.000000,50000.000000,50000.000000,50000.000000,50000.000000
mean,125000.500000,3.146102,8.427752,2.80026,11.209271,3.505360,4.270760,3.349299,75.798125
std,14433.901067,0.478854,8.269490,1.18802,5.156426,1.820812,2.144066,0.495673,13.281626
min,100001.000000,1.183000,0.000000,1.00000,1.000000,1.000000,1.000000,1.000000,10.780000
25%,112500.750000,2.834000,2.390000,2.00000,7.560000,2.000000,3.000000,3.023750,66.820000
50%,125000.500000,3.210000,5.800000,3.00000,11.180000,3.000000,4.000000,3.421000,76.000000
75%,137500.250000,3.521000,11.720000,4.00000,14.710000,5.000000,6.000000,3.749000,85.190000
max,150000.000000,3.998000,40.000000,5.00000,35.860000,10.000000,10.000000,4.000000,100.000000


### DATA CLEANING

In [7]:
df.isnull().sum()

Student_ID                    0
Major_Category                0
Year_of_Study                 0
Pre_Semester_GPA              0
Weekly_GenAI_Hours            0
Primary_Use_Case              0
Prompt_Engineering_Skill      0
Tool_Diversity                0
Paid_Subscription             0
Traditional_Study_Hours       0
Perceived_AI_Dependency       0
Institutional_Policy          0
Anxiety_Level_During_Exams    0
Post_Semester_GPA             0
Skill_Retention_Score         0
Burnout_Risk_Level            0
dtype: int64

**Observation:**
No missing values were found in any column of the dataset.Therefore,no missing-value treatment was required.

In [8]:
df.duplicated().sum()

np.int64(0)

**Observation:**  
No duplicate rows were found in the dataset. Therefore, no duplicate records were removed.

In [9]:
df.dtypes

Student_ID                      int64
Major_Category                 object
Year_of_Study                  object
Pre_Semester_GPA              float64
Weekly_GenAI_Hours            float64
Primary_Use_Case               object
Prompt_Engineering_Skill       object
Tool_Diversity                  int64
Paid_Subscription                bool
Traditional_Study_Hours       float64
Perceived_AI_Dependency         int64
Institutional_Policy           object
Anxiety_Level_During_Exams      int64
Post_Semester_GPA             float64
Skill_Retention_Score         float64
Burnout_Risk_Level             object
dtype: object

**Observation:**  
The data types of all columns were checked and found to be appropriate for the dataset. No data type conversion was required.

<div style="
    text-align: center;
    margin: 35px 0 25px 0;
">
    <h1 style="
        font-size: 28px;
        letter-spacing: 2px;
        color:light blue;
        margin: 0;
    ">
        BASIC DATA ANALYSIS
    </h1>
    <div style="
        width: 55px;
        height: 2px;
        background: white;
        margin: 12px auto 0 auto;
    "></div>
</div>

#### Major Category Distribution

In [10]:
df["Major_Category"].value_counts()

Major_Category
STEM          15059
Business      12538
Humanities     9994
Medical        6476
Arts           5933
Name: count, dtype: int64

**Observation:**  
STEM has the highest number of students (15,079), while Arts has the lowest number of students (5,933). Business, Humanities, and Medical fall between these two categories.

#### Primary Use Case Distribution

In [12]:
df["Primary_Use_Case"].value_counts()

Primary_Use_Case
Debugging/Troubleshooting    12295
Copywriting/Drafting         12011
Ideation                     10721
Summarizing_Reading           8633
Direct_Answer_Generation      6340
Name: count, dtype: int64

**Observation:**  
Debugging or Troubleshooting is the most common AI use case, while Direct Answer Generation is the least common.

#### AI Usage Across Years Of Study

In [13]:
df.groupby("Year_of_Study")["Weekly_GenAI_Hours"].mean()

Year_of_Study
Freshman     8.446962
Graduate     8.406052
Junior       8.376325
Senior       8.459251
Sophomore    8.446249
Name: Weekly_GenAI_Hours, dtype: float64

**Observation:**  
AI usage is fairly similar across all years of study, with Senior students showing the highest average usage.

#### AI Usage vs Traditional Study Hours

In [6]:
# add new column(AI_Usage_Level)
df["AI_Usage_Level"]=pd.cut(
    df["Weekly_GenAI_Hours"],
    bins=[0,5,10,float("inf")],
    labels=["Low","Medium","High"]
)


In [9]:
df[["Weekly_GenAI_Hours",
    "AI_Usage_Level"]].head()

,Weekly_GenAI_Hours,AI_Usage_Level
0,23.31,High
1,1.12,Low
2,21.26,High
3,1.82,Low
4,9.29,Medium


In [10]:
df.groupby("AI_Usage_Level")["Traditional_Study_Hours"].mean()

AI_Usage_Level
Low       11.803943
Medium    11.373168
High      10.195338
Name: Traditional_Study_Hours, dtype: float64

**Observation:**  
Low AI users have the highest average traditional study hours (11.80 hours), while high AI users have the lowest (10.19 hours). This shows a decreasing pattern in traditional study hours as AI usage increases.

#### Pre-Semester GPA vs Post-Semester GPA

In [11]:
df[["Pre_Semester_GPA","Post_Semester_GPA"]].mean()

Pre_Semester_GPA     3.146102
Post_Semester_GPA    3.349299
dtype: float64

**Observation:**  
The average GPA increased from 3.14 before the semester to 3.34 after the semester.

#### Burnout Risk Level Distribution

In [12]:
df["Burnout_Risk_Level"].value_counts()

Burnout_Risk_Level
Medium    21144
Low       16369
High      12487
Name: count, dtype: int64

**Observation:**  
Most students fall under the Medium burnout risk level (21,144), followed by Low (16,369) and High (12,487). This shows that the majority of students in the dataset have a medium level of burnout risk.

#### AI Usage and Burnout Risk

In [13]:
pd.crosstab(df["AI_Usage_Level"],
            df["Burnout_Risk_Level"])

Burnout_Risk_Level,High,Low,Medium
AI_Usage_Level,,,
Low,2366,10521,9677
Medium,2310,4038,5825
High,7810,1795,5630


**Observation:**  
Low AI users mostly fall under low burnout risk, while high AI users have the highest number of students in the high burnout risk category. The dataset shows an increasing pattern between AI usage level and burnout risk level.

#### Paid AI Subscription Usage

In [14]:
df["Paid_Subscription"].value_counts()

Paid_Subscription
False    28846
True     21154
Name: count, dtype: int64

**Observation:**  
Most students do not have a paid AI subscription. Out of 50,000 students, 28,846 are non-subscribers, while 21,154 have a paid subscription.

#### Prompt Engineering Skill Distribution

In [15]:
df["Prompt_Engineering_Skill"].value_counts()

Prompt_Engineering_Skill
Beginner        18495
Intermediate    17696
Advanced        13809
Name: count, dtype: int64

**Observation:**  
Most students have a Beginner level of prompt engineering skill (18,495), followed by Intermediate (16,696) and Advanced (13,809). This shows that beginner-level prompt engineering skill is the most common among the students in the dataset.

#### AI Tool Diversity Distribution

In [16]:
df["Tool_Diversity"].value_counts().sort_index()

Tool_Diversity
1     8141
2    11830
3    16926
4     8081
5     5022
Name: count, dtype: int64

**Observation:**  
Most students use 3 different AI tools (16,926), while the fewest students use 5 tools (5,022). The dataset shows that using 3 AI tools is the most common level of tool diversity among students.

#### Institutional AI Policy Distribution

In [17]:
df["Institutional_Policy"].value_counts()

Institutional_Policy
Allowed_With_Citation    25224
Actively_Encouraged      14988
Strict_Ban                9788
Name: count, dtype: int64

**Observation:**  
Most students are from institutions that allow AI use with citation. This shows that AI use is more commonly permitted than strictly banned in the dataset.

#### AI Usage Level vs Exam Anxiety

In [7]:
df.groupby("AI_Usage_Level")["Anxiety_Level_During_Exams"].mean()

AI_Usage_Level
Low       3.846969
Medium    4.068841
High      5.060387
Name: Anxiety_Level_During_Exams, dtype: float64


**Observation:**
AI usage level increases, the average exam anxiety score also increases. High AI users have the highest average anxiety score.

#### AI Usage Level vs Skill Retention Score

In [8]:
df.groupby("AI_Usage_Level")["Skill_Retention_Score"].mean()

AI_Usage_Level
Low       75.978470
Medium    77.078770
High      74.504435
Name: Skill_Retention_Score, dtype: float64

**Observation:**  
There is no clear pattern between AI usage level and skill retention score. Higher AI usage does not show a consistent increase or decrease in skill retention.

### SAVING THE FINAL DATASET

In [18]:
df.to_csv("final_ai_impact_students.csv")

<div style="
    background-color: #f8f9fa;
    border-left: 5px solid #333;
    border-radius: 10px;
    padding: 20px 25px;
    margin: 25px 0;
    box-shadow: 0 2px 8px rgba(0,0,0,0.08);
">
       <h2 style="
        text-align: center;
        font-size: 24px;
        margin-bottom: 15px;
        color: #222;
    ">
Conclusion
</h2>

<p style="
    text-align: justify;
    font-size: 16px;
    color: #333;
">
This project explored the use of Generative AI among students and its
association with their study habits, academic performance and academic
experience. The analysis showed that students use AI for different purposes,
with debugging and troubleshooting being the most common use case.
</p>

<p style="
    text-align: justify;
    font-size: 16px;
    color: #333;
">
I compared the average Weekly Genarative AI Hours with the average Traditionl Study Hours.
From this comparison, I found that the time spent on traditional learning is greater.

<p style="
    text-align: justify;
    font-size: 16px;
    color: #333;
">
The analysis also showed that students with different levels of AI usage
displayed different patterns in traditional study hours and burnout risk.
The average GPA increased from 3.14 before the semester to 3.34 after the
semester, while the average perceived AI dependency score was 3.50.
</p>

<p style="
    text-align: justify;
    font-size: 16px;
    color: #333;
">
Overall, the project demonstrates how data analysis can be used to understand
patterns in students' AI usage and their academic experience. The findings
provide useful insights into how Generative AI is being used by students
within the given dataset.
</p>

</div>